### PR TITLE
couchbase-shell: init at 1.0.0

### PR DESCRIPTION
**Pull request by petrkozorezov**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/co/couchbase-shell/package.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    updateScript = nix-update-script { };
+  };
+
+  meta = with lib; {
+    description = "Couchbase Shell (cbsh) is a modern, productive and fun shell for Couchbase Server and Cloud.";
+    homepage = "https://couchbase.sh/";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ petrkozorezov ];
+    mainProgram = "cbsh";
+  };
```
Comment (romner-set) [inline pkgs/by-name/co/couchbase-shell/package.nix:58]: What platforms has this been tested on? Set them in `meta.plaforms`. Nested `with`s should also not be used.

```suggestion
  meta = {
    description = "Shell for Couchbase Server and Cloud";
    homepage = "https://couchbase.sh/";
    license = lib.licenses.asl20;
    maintainers = with lib.maintainers; [ petrkozorezov ];
    mainProgram = "cbsh";
  };
```

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -16487,6 +16487,13 @@
     githubId = 28323;
     name = "Peter Simons";
   };
+  petrkozorezov = {
+    email = "petr.kozorezov@gmail.com";
+    github = "petrkozorezov";
+    githubId = 645017;
+    name = "Petr Kozorezov";
+    keys = [ { fingerprint = "7F1A 353D 3D6D 9CEF 63A9  B5C6 699F 32D5 9999 7C90"; } ];
```
Comment (romner-set) [inline maintainers/maintainer-list.nix:16495]: I can't seem to receive the key properly, `new key but contains no user ID - skipped`.

**Review by maevii — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 2 above)
Comment (petrkozorezov) [reply maintainers/maintainer-list.nix:16495]: It was an issue with key email verification.
Should work now.

**Comment by petrkozorezov** [comment]
I applied suggestions and force pushed it.

File: pkgs/by-name/co/couchbase-shell/package.nix [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  meta = {
+    description = "Shell for Couchbase Server and Cloud";
+    homepage = "https://couchbase.sh/";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ petrkozorezov ];
+    mainProgram = "cbsh";
+    platforms = [
+      "x86_64-linux"
+      "aarch64-darwin"
+    ];
```
Comment (romner-set) [inline pkgs/by-name/co/couchbase-shell/package.nix:61]: Seems to work fine on aarch64-linux as well, presumably it works on linux & darwin on any architecture.
```suggestion
    platforms = lib.platforms.linux ++ lib.platforms.darwin;
```

**Review by maevii — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 2 above)
Comment (romner-set) [reply maintainers/maintainer-list.nix:16495]: Works fine now, thanks.

**Comment by romner-set** [comment]
Result of `nixpkgs-review pr 347727` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>couchbase-shell</li>
  </ul>
</details>

**Review by maevii — APPROVED** [review]
Alright, LGTM. Good work.

##### Reviewed points

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [x] package builds on aarch64-linux
- [x] executables tested on x86_64-linux
- [x] executables tested on aarch64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] `meta.mainProgram` is set, if applicable.
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden

**Comment by petrkozorezov** [comment]
Removed darwin support due to broken ofborg build tests. Will investigate it late.

**Comment by Scrumplex** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 347727`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>couchbase-shell</li>
  </ul>
</details>

**Review by Scrumplex — APPROVED** [review]
Builds and runs

**Pull request merged**